server/src/test/java/org/elasticsearch/action/support/PlainActionFutureTests.java

```diff
@@ -142,6 +145,34 @@ public void testCancelException() {
         assertPropagatesInterrupt(() -> future.actionGet(10, TimeUnit.SECONDS));
     }
 
+    public void testAssertCompleteAllowedAllowsConcurrentCompletesFromSamePool() {
+        assumeTrue("Assertions need to be enabled for assertCompleteAllowed to be invoked", Assertions.ENABLED);
```

Comment:
Is this assumption necessary any more?

Reply:
Yes, because
```java
@Override
    public void onFailure(Exception e) {
        assert assertCompleteAllowed();  // <----- only tested when assertions enabled
        ...
```

Reply:
Sure, but the test is still valid either way AFAICT. I mean most tests make stronger checks if assertions are enabled, but that doesn't mean we should suppress them all like this.

Reply:
Yes, fair. I guess I was over-fitting the test to the issue that was fixed (rather than implementing a generic "can complete futures concurrently"). Have removed.